### ISSUE TITLE
[AIR 51.0.1.5][Android] Video playback doesn't work (`NetStream.Play.FileStructureInvalid` error)

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#3380]
## Problem Description
Video playback doesn't work with AIR 51.0.1.5 for Android. `NetStream.Play.FileStructureInvalid` error occurs.
Maybe it is side effect of https://github.com/airsdk/Adobe-Runtime-Support/issues/3353

Tested with latest AIR 51.0.1.5 with multiple different Android devices, different OS versions, architectures, with different applications and videos.
Same issue in all cases.
There is no such issue using AIR 51.0.1.4 or below.

Related issues:
https://github.com/airsdk/Adobe-Runtime-Support/issues/3115
https://github.com/airsdk/Adobe-Runtime-Support/issues/2934
https://github.com/airsdk/Adobe-Runtime-Support/issues/2402
https://github.com/airsdk/Adobe-Runtime-Support/issues/2125
https://github.com/airsdk/Adobe-Runtime-Support/issues/1939
https://github.com/airsdk/Adobe-Runtime-Support/issues/1159
https://github.com/airsdk/Adobe-Runtime-Support/issues/224
https://github.com/airsdk/Adobe-Runtime-Support/issues/211
https://github.com/airsdk/Adobe-Runtime-Support/issues/155
https://github.com/airsdk/Adobe-Runtime-Support/issues/139
https://github.com/airsdk/Adobe-Runtime-Support/issues/93
https://github.com/airsdk/Adobe-Runtime-Support/issues/82

## Steps to Reproduce
Launch application with code below with any Android device.

Application example with sources attached.
[air51_android_video_filestructureinvalid_bug.zip](https://github.com/user-attachments/files/16401149/air51_android_video_filestructureinvalid_bug.zip)

```
package {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	
	public class AIR51AndroidVideoFileStructureInvalidBug extends Sprite {
		private var nc:NetConnection;
		private var ns:NetStream;
		private var video:Video;
		
		public function AIR51AndroidVideoFileStructureInvalidBug() {
			addEventListener(Event.ADDED_TO_STAGE, init);
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			
			startVideo();
		}
	
		private function startVideo():void {
			removeChildren();
			
			if (ns != null){
				video.attachNetStream(null);
				ns.removeEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				ns.dispose();
			}
		
			if (video != null){
				video.clear();
			}
			
			video = new Video(640, 480);
			addChild(video);
			
			if (nc != null){
				nc.removeEventListener(NetStatusEvent.NET_STATUS, ncHandler);
				nc.close();
			}
			
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
		}
		
		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				ns = new NetStream(nc);
				ns.checkPolicyFile = false;
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);				
				ns.client = {onMetaData:getMeta, onPlayStatus:onPlayStatus};
				
				video.attachNetStream(ns);
				ns.play("video.mp4");
			}
		}

		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);//NetStream.Play.FileStructureInvalid
			if (e.info.code == "NetStream.Play.Stop"){
				startVideo();
			}
		}

		private function getMeta(mdata:Object):void { }
	
		private function onPlayStatus(infoObject:Object):void {
			trace("onPlayStatus", infoObject.code);
		}
	}
}
```

**Actual Result:**
Video playback doesn't work.
In traces you will see:
`NetStream.Play.FileStructureInvalid`

**Expected Result:**
Video playback.

## Known Workarounds
none
*Use AIR 51.0.1.4 or below

**Comment by itlancer** [comment]
Issue still exists with latest AIR 51.1.1.1.

**Comment by ajwfrost** [comment]
You're right in that this is a side-effect of #3353. The issue being, the operating system call to `lseek64` is returning an incorrect / large value for the mp4 file. Having returned the correct value several times.. 

It's not returning any failure value, it's just giving us the wrong value. Not entirely sure why! The file is coming from the APK though so it may be complicated by that - which is an area we were looking to rework entirely at some point.

We can probably work around this with a mechanism that combines the old and new systems so that we can continue to support larger files but don't cause this regression..

thanks

**Comment by itlancer** [comment]
Fixed with latest AIR 51.1.1.2.
Thanks!
